### PR TITLE
Add C# standards

### PR DESCRIPTION
**Pull request by Amerdrix**
A start implementing some C# standards.

### CODE REVIEW

File: cSharp.md [hunk 1]
```diff
@@ -0,0 +1,25 @@
+# C\#
+
+## Code Style
+
+C# Projects have style enforced by [StyleCop] (https://stylecop.codeplex.com/)
+
+Notable rules enfoced are:
```
Comment (jordanmaguire) [inline cSharp.md]: enforced

**Comment by jordanmaguire** [comment]
such a wow :+1:

File: cSharp.md (hunk 1 above)
Comment (Amerdrix) [inline cSharp.md]: Nah, I wanted `enfoced` :)

Comment (jordanmaguire) [inline cSharp.md]: oh well I can't foce you to change it

File: cSharp.md [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+C# Projects have style enforced by [StyleCop] (https://stylecop.codeplex.com/)
+
+Notable rules enforced are:
+
+* Naming conventions. 
+ * Properties are UpperCamelCased
+ * Classes and Structs are UpperCamelCased
+ * Fields are _underscoreLowerCamelCased
+ * Interfaces are UpperCamelCased and prefixed with I  (ISomeInterface)
+ * Members be orederd Static to Instance then Public to Private
```
Comment (pzi) [inline cSharp.md]: ordered

File: cSharp.md [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+
+* A target must be specifed for all non local access. This means `this` must be used for instance methods / properties / fields, and `SomeClassName` must be specified for static methods.
```
Comment (pzi) [inline cSharp.md]: missing a subject: non local access what? :)

**Pull request merged**